libraries/render-utils/src/text/Font.cpp

```diff
@@ -388,10 +370,71 @@ void Font::buildVertices(Font::DrawInfo& drawInfo, const QString& str, const glm
             advance.x += _spaceWidth;
         }
     }
+
+    std::vector<QuadBuilder> quadBuilders;
+    quadBuilders.reserve(glyphsAndCorners.size());
+    {
+        int i = glyphsAndCorners.size() - 1;
+        while (i >= 0) {
```

Comment:
then, we loop over all the pairs backwards.  we take the first Glyph (the end of the line), use it to compute the spacing on the right, then adjust all of the characters on the line appropriately.  this will build a list of QuadBuilders in reverse order.  we detect when we go to a previous line to recalculate the right spacing